runtime/lib/formatter/PropelObjectFormatter.php

```diff
@@ -39,16 +39,21 @@ public function format(PDOStatement $stmt)
             $pks = array();
             $objectsByPks = array();
             while ($row = $stmt->fetch(PDO::FETCH_NUM)) {
-                $object = $this->getAllObjectsFromRow($row);
-                $pk = $object->getPrimaryKey();
 
                 if (false === Propel::isInstancePoolingEnabled()) {
+
+                    list($object, $col) = call_user_func(array($this->peer, 'populateObject'), $row);
+                    $pk = $object->getPrimaryKey();
+
                     if (isset($objectsByPks[$pk])) {
                         $this->mainObject = $objectsByPks[$pk];
-                        $object = $this->getAllObjectsFromRow($row);
                     }
+                    $object = $this->getAllObjectsFromRow($row);
```

Comment:
Not really. The issue is that the method `getAllObjectsFromRow` is called twice when pooling is disabled and is causing issues during hydration when joins (->joinWith() ) are used. What I discovered is, that it needs the primary object to get the primary key in order to set the 'mainObject', but by calling the "getAllObjectsFromRow" it does too much, reason why I extract the first line of that method which only populates the "main object" and not the joined/nested objects. Only calling `getAllObjectsFromRow` once (as when pooling is enabled) solves everything.

Maybe there is a better solution, but I didn't want to make it to fancy as in propel2 it has already be done differently.